examples/pennylane_run_doubly_stochastic.py

```diff
@@ -0,0 +1,312 @@
+r"""
+Doubly stochastic gradient descent
+==================================
+
+In this tutorial we investigate and implement the doubly stochastic gradient descent
+paper from `Ryan Sweke et al. (2019) <https://arxiv.org/abs/1910.01155>`__. In this paper,
+it is shown that quantum gradient descent, where a finite number of measurement samples
+(or *shots*) are used to estimate the gradient, is a form of stochastic gradient descent.
+Furthermore, if the optimization involves a linear combination of expectation values
+(such as VQE), sampling from the terms in this linear combination can further reduce required
+resources, allowing for "doubly stochastic gradient descent".
+
+Background
+----------
+
+In classical machine learning, `stochastic gradient descent
+<https://en.wikipedia.org/wiki/Stochastic_gradient_descent>`__ is a common optimization strategy
+where the standard gradient descent parameter update rule,
+
+.. math:: \theta^{(t+1)} = \theta^{(t)} - \eta \nabla \mathcal{L}(\theta^{(t)}),
+
+is modified such that
+
+.. math:: \theta^{(t+1)} = \theta^{(t)} - \eta g^{(t)}(\theta^{(t)})
+
+where :math:`\eta` is the step-size, and :math:`\{g^{(t)}(\theta)\}` is a sequence of random
+variables such that
+
+.. math:: \mathbb{E}[g^{(t)}(\theta)] = \nabla\mathcal{L}(\theta).
+
+In general, stochastic gradient descent is preferred over standard gradient
+descent for several reasons:
+
+1. Samples of the gradient estimator :math:`g^{(t)}(\theta)` can typically
+   be computed much more efficiently than :math:`\mathcal{L}(\theta)`,
+
+2. Stochasticity helps avoid local minima and saddle points,
+
+3. Convergence is guaranteed to be on par with regular gradient descent.

```

Comment:
```suggestion
3. One can prove rigorous convergence guarantees in simplified settings.
```

Not sure if this is catchy enough?